modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/api/IgniteSqlImpl.java

```diff
@@ -327,11 +322,12 @@ public <T> CompletableFuture<AsyncResultSet<T>> executeAsync(
 
     private CompletableFuture<AsyncResultSet<SqlRow>> executeAsyncInternal(
             @Nullable Transaction transaction,
-            String query,
-            int pageSize,
+            Statement statement,
             @Nullable Object... arguments
     ) {
-        assert pageSize > 0 : pageSize;
+        assert statement.pageSize() >= 0 : statement.pageSize();
```

Comment:
Maybe It is better to make it impossible to create invalid statements ? (negative page size, negative timeout etc.)

Reply:
> Maybe It is better to make it impossible to create invalid statement
Thanks, it's impossible now. 
Validation is done in `StatementImpl`.
I just kept assertion that pageSize is positive here.